Update PPP + About Us (Partial) (#341)
AY2223S2-CS2103T-W14-2/tp: docs/team/russelltankaimin.md

### Project: EduMate

`EduMate` is a desktop address book application used to help NUS students maintain both their social and academic life by lowering the barriers to meet up and also make new friends within modules and school. 
The user interacts with it using a CLI, and it has a GUI created with JavaFX. It is written in Java, and has about 30 kLoC.

Given below are my contributions to the project.
* **Code Contributed to team-based tasks**:
  * **Basic User Profile** [#7](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/7)
    * What it does: Separate User Profile from a typical contact.
    * Justification: Personalise settings for the user of the application.
  * **Overhaul of `EduMate` From AB3** [#7](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/7), [#32](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/32):
    * What it does: Gives a basic structure of `EduMate` to work with.
    * Justification: Fits into requirements of `EduMate` through adaptation of code.

>   * **Spearhead use of `ContactIndex` and `IndexHandler`** [#124](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/124) Testing: [#134](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/134)

Reviewer: ```suggestion
  * **Spearhead the use of `ContactIndex` and `IndexHandler`** [#124](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/124) Testing: [#134](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/134)
```

    * What it does: Gives a unique identifier for each contact added to `EduMate`, replacing `Index` entirely.
    * Justification: Use of `Index` is too rigid as it obtains a `Person` from their index in `ObservableList` only.
    * Highlight: `IndexHandler` can help assign a `ContactIndex` to a newly-added `Person`

>       and that contacts could be queried by their `ContactIndex` instead of their index in `ObservableList`.

Reviewer: Grammar.

Author: which one uh

Reviewer: "contacts".

  * **Person Card UI changes** [#134](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/134)

>     * Let each Person Card's index number to display `ContactIndex` instead of just their relative index in `ObservableList`.

Reviewer: ```suggestion
    * Let each Person Card's index number display `ContactIndex` instead of just their relative index in `ObservableList`.
```

  * **`HourBlock`, `TimeBlock`, `TimePeriod`, `Day`** [#144](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/144), [#193](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/193) Testing: [#144](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/144), [#193](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/193)
    * What it does: Represents periods of a time in a `Timetable`.
    * Justification: Abstracts out a period of time for the `TimingRecommender`.
    * Highlight: Built using [Joda-Time](https://www.joda.org/joda-time/index.html)
  * **MathUtil** [#193](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/193)
    * What it does: Uses mathematical tools which assists `TimingRecommender` to perform its functions.
    * Justification: Can exploit mathematical properties and structure for cleaner implementation of code.
    * Highlights: Cartesian Product, Indexing.
  * **TimeUtil** [#144](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/144), [#193](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/193) Testing: [#193](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/193)

>     * What it does: Utility functions which help to time-based classes (`HourBlock`, `Timetable` etc) to perform its functionalities.

Reviewer: Grammar does not check out here.

    * Justification: These methods do not necessarily appear in some time-based classes due to OOP.
    * Highlights: Clash Checks, consecutive `TimeBlock`/`HourBlock` checks, merging suitable `TimeBlocks` or `HourBlocks` to form a single `TimePeriods`.
  * **`Timetable` Class** [#144](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/144), [#193](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/193) Testing: [#144](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/144)

>     * What it does: Represents a contact's or your timetable.

Reviewer: "your" to be removed.

    * Justification: To display the schedule of the `Person` or `User`
  * **`TimingRecommender`** [#144](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/144), [#204](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/204)
    * What it does: Looks through all participant's timetables and find time slots that every participant is free.

>     * Justification: Automated tool does the job faster than NUS students messaging each other when is a good time to meet.

Reviewer: Could be phrased better.
```suggestion
    * Justification: Using an automated tool to schedule meetups is more efficient compared to relying on NUS students to manually message each other to find a suitable meeting time.
```

>   * **`meet` Command and `TimingRecommender` integration** [#204](https://github.com/AY2223S2-CS2103T-W14-2/tp/pull/204)

Reviewer: Earlier, the `View` command was capitalised. This should be as well.
